chanzuckerberg/idseq-web
# Show genus tooltip on hover in heatmap

# Description

This adds a tooltip on hover of the new genus groups in the heatmap according to https://chanzuckerberg.invisionapp.com/d/main#/console/18553478/385796250/inspect. See https://github.com/chanzuckerberg/idseq-web/pull/2675. 

![image](https://user-images.githubusercontent.com/28797/67806909-fec28780-fa50-11e9-8815-ad45160c28d0.png)

# Notes

* This is most useful for viruses which don't have their genus in their species name
* The tooltip is fixed width to make it possible to center it
* The width appears wider than in the design because the width of the column is of the longest row labels
* It might be nice to add a delay effect to only show after 1 second
* I didn't use the built in BasicPopup because it doesn't play nice with SVG
* This took a lot of tweaking to get right!

# Tests

* Load heatmap, hover over labels
* See correct genus

* Try unknown genus

* Try zooming in and out and the same thing
